API Migrate SilverStripeNagivator

GuySartorelli opened the pull request:
These `SilverStripeNavigatorItem` subclasses are useful for `Versioned` objects without requiring `silverstripe/cms`
I'm not adding them directly to `silverstripe/versioned` because that module has no dependency on `silverstripe/admin` - nor should it.

See also https://github.com/silverstripe/silverstripe-cms/pull/2761

## Parent issue
- https://github.com/silverstripe/silverstripe-admin/issues/1339

GuySartorelli committed:
API Migrate SilverStripeNagivator

GuySartorelli commented:
Needs https://github.com/silverstripe/silverstripe-admin/pull/1340 to be merged before it will go green.

sabina-talipova commented:
Could you fix lint issues: https://github.com/silverstripe/silverstripe-versioned-admin/actions/runs/3049723059/jobs/5197764881, please.

GuySartorelli committed:
MNT Fix PHPCS linting issues

GuySartorelli commented:
Done - I've updated the linting rules to allow the class names, since these are the same class names that were in use in the cms module.
I've also added some items to the `.gitignore` file which shouldn't be committed, but are sometimes present during development.

sabina-talipova commented on tests/Navigator/SilverStripeNavigatorTest.php:
```diff
@@ -0,0 +1,221 @@
+
+use SilverStripe\Admin\Navigator\SilverStripeNavigator;
+use SilverStripe\Admin\Navigator\SilverStripeNavigatorItem_Unversioned;
+use SilverStripe\Versioned\Navigator\SilverStripeNavigatorItem_ArchiveLink;
+use SilverStripe\Versioned\Navigator\SilverStripeNavigatorItem_LiveLink;
+use SilverStripe\Versioned\Navigator\SilverStripeNavigatorItem_StageLink;
```
```suggestion
use SilverStripe\VersionedAdmin\Navigator\SilverStripeNavigatorItem_ArchiveLink;
use SilverStripe\VersionedAdmin\Navigator\SilverStripeNavigatorItem_LiveLink;
use SilverStripe\VersionedAdmin\Navigator\SilverStripeNavigatorItem_StageLink;
```

ChloeSartorelli replied:
Done

sabina-talipova commented on tests/Navigator/SilverStripeNavigatorTest.php:
```diff
@@ -0,0 +1,221 @@
+use SilverStripe\Versioned\Navigator\SilverStripeNavigatorItem_ArchiveLink;
+use SilverStripe\Versioned\Navigator\SilverStripeNavigatorItem_LiveLink;
+use SilverStripe\Versioned\Navigator\SilverStripeNavigatorItem_StageLink;
+use SilverStripe\Dev\SapphireTest;
```
```suggestion
use SilverStripe\Dev\SapphireTest;
use SilverStripe\VersionedAdmin\Tests\Navigator\UnstagedRecord;
use SilverStripe\VersionedAdmin\Tests\Navigator\UnversionedRecord;
use SilverStripe\VersionedAdmin\Tests\Navigator\VersionedRecord;
```

ChloeSartorelli replied:
Done
